src/routes/spaces/address-books.controller.spec.ts

```diff
@@ -420,6 +448,115 @@ describe('AddressBooksController', () => {
+        .put(`/v1/spaces/${spaceId}/address-book`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .send({
+          items: [
+            {
+              name: addressBookItem2.name,
+              address: addressBookItem2.address,
+              chainIds: addressBookItem2.chainIds,
+            },
+          ],
+        })
+        .expect(429)
+        .expect('Rate limit reached');
+    });
+
+    it('should return a 400 if the limit is reached', async () => {
+      const app = await initApp({ maxUpsertionsPerUser: 1 });
+      const { spaceId, accessToken } = await createSpace();
+      const addressBookItem1 = addressBookItemBuilder().build();
+      const addressBookItem2 = addressBookItemBuilder().build();
+
+      await request(app.getHttpServer())
+        .put(`/v1/spaces/${spaceId}/address-book`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .send({
+          items: [
+            {
+              name: addressBookItem1.name,
+              address: addressBookItem1.address,
+              chainIds: addressBookItem1.chainIds,
+            },
+          ],
+        })
+        .expect(200);
+
+      await request(app.getHttpServer())
+        .put(`/v1/spaces/${spaceId}/address-book`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .send({
+          items: [
+            {
+              name: addressBookItem2.name,
+              address: addressBookItem2.address,
+              chainIds: addressBookItem2.chainIds,
+            },
+          ],
+        })
+        .expect(400)
+        .expect({
+          message:
+            'This Space only allows a maximum of 1 Address Book Items. You can only add up to 0 more.',
+          error: 'Bad Request',
+          statusCode: 400,
+        });
+    });
+
+    it('should return a 400 if the request has more than the limit', async () => {
+      const app = await initApp({ maxUpsertionsPerUser: 1 });
+      const { spaceId, accessToken } = await createSpace();
+      const addressBookItem1 = addressBookItemBuilder().build();
+      const addressBookItem2 = addressBookItemBuilder().build();
+
+      await request(app.getHttpServer())
+        .put(`/v1/spaces/${spaceId}/address-book`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .send({
+          items: [
+            {
+              name: addressBookItem1.name,
+              address: addressBookItem1.address,
+              chainIds: addressBookItem1.chainIds,
+            },
+            {
+              name: addressBookItem2.name,
+              address: addressBookItem2.address,
+              chainIds: addressBookItem2.chainIds,
+            },
+          ],
+        })
+        .expect(400)
+        .expect({
+          message:
+            'This Space only allows a maximum of 1 Address Book Items. You can only add up to 1 more.',
+          error: 'Bad Request',
+          statusCode: 400,
+        });
+    });
```

Comment:
Whilst not explicitly related to rate limiting, I noticed that test coverage was missing and added it here.